plugins/lua/sort/places.lua

```diff
@@ -33,9 +33,45 @@ local language_name_types = {
     [df.language_name_type.SymbolFood] = 'Inn Tavern',
     [df.language_name_type.Temple] = 'Temple',
     [df.language_name_type.Hospital] = 'Hospital',
-    [df.language_name_type.Guildhall] = 'Guildhall'
+    [df.language_name_type.Guildhall] = 'Guildhall',
+    [df.language_name_type.Library] = 'Library'
 }
 
+local function add_spheres(hf, spheres)
+    if not hf then return end
+    for _, sphere in ipairs(hf.info.spheres.spheres) do
+        spheres[sphere] = true
+    end
+end
+
+local function stringify_spheres(spheres)
+    local strs = {}
+    for sphere in pairs(spheres) do
+        table.insert(strs, df.sphere_type[sphere])
+    end
+    return table.concat(strs, ' ')
+end
+
+local function get_religion_string(religion_id, religion_type)
```

Comment:
I would've just un-localized the existing `get_religion_string` but want to return slightly different things (namely 'Temple' when `religion_type` is `None (-1)`). Let me know if there's a better way to reuse this code.